src/Writer/AbstractWriter.php

```diff
@@ -77,6 +79,14 @@ public function __construct($options = null)
         }
 
         if (is_array($options)) {
+            if (isset($options['filter_manager']) && $options['filter_manager'] instanceof FilterPluginManager) {
+                $this->setFilterPluginManager($options['filter_manager']);
+            }
+
+            if (isset($options['formatter_manager']) && $options['formatter_manager'] instanceof FormatterPluginManager) {
```

Comment:
mmm, how about skipping instanceof check and letting it to hard fail on setter typehint if unexpected value is provided? That will close off potential for pretty big wtf moment.